packages/webview_flutter/webview_flutter_android/android/src/main/java/io/flutter/plugins/webviewflutter/WebChromeClientHostApiImpl.java

```diff
@@ -29,28 +38,96 @@ public class WebChromeClientHostApiImpl implements WebChromeClientHostApi {
   private final InstanceManager instanceManager;
   private final WebChromeClientCreator webChromeClientCreator;
   private final WebChromeClientFlutterApiImpl flutterApi;
+  private Context context;
 
   /**
    * Implementation of {@link WebChromeClient} that passes arguments of callback methods to Dart.
    */
   public static class WebChromeClientImpl extends SecureWebChromeClient {
     private final WebChromeClientFlutterApiImpl flutterApi;
     private boolean returnValueForOnShowFileChooser = false;
+    private View mFullscreenView;
+    private Context context;
 
     /**
      * Creates a {@link WebChromeClient} that passes arguments of callbacks methods to Dart.
      *
      * @param flutterApi handles sending messages to Dart
+     * @param context the Activity or application context
      */
-    public WebChromeClientImpl(@NonNull WebChromeClientFlutterApiImpl flutterApi) {
+    public WebChromeClientImpl(
+        @NonNull WebChromeClientFlutterApiImpl flutterApi, @NonNull Context context) {
       this.flutterApi = flutterApi;
+      this.context = context;
     }
 
     @Override
     public void onProgressChanged(@NonNull WebView view, int progress) {
       flutterApi.onProgressChanged(this, view, (long) progress, reply -> {});
     }
 
+    @Override
+    public void onShowCustomView(View view, WebChromeClient.CustomViewCallback callback) {
```

Comment:
Since this provides the `View`, I want to try and pass the view back as a `Widget`. I think we can try to allow the user to handle the transition to and from a fullscreen view. That way our handling of system ui visibility doesn't contradict the use of https://api.flutter.dev/flutter/services/SystemChrome/setEnabledSystemUIMode.html. 

I have a generator for wrapping native classes, so I can update the PR with the `View` and `WebChromeClient.CustomViewCallback` wrapped. Then you can handle the Dart code.

Reply:
Sounds like a sound plan. If you would like to update the PR, please go ahead!

Reply:
Forgot to come back to this. I updated the PR with `WebChromeClient` wrapper updated with the `onShowCustomView` and `onHideCustomView`. You should be able to add a method in `AndroidWebViewController` that sets callbacks to receive these events. And you should be able to pass a PlatformView `Widget` the same way `AndroidWebViewWidget.build` does.